app/eventCreation/index.tsx

```diff
@@ -0,0 +1,174 @@
+import {
+  View,
+  KeyboardAvoidingView,
+  Platform,
+  Keyboard,
+  Pressable,
+} from "react-native";
+import { Button, YStack, SizableText, TextArea, XStack, ScrollView } from "tamagui";
+import React, { useState } from "react";
+import { FormField } from "@/components/FormField";
+import { useForm } from "react-hook-form";
+import { router } from "expo-router";
+import { useEventCreation } from "@/hooks/use-event-creation";
+import { useUser } from "@/hooks/use-user";
+import DateTimePickerModal from "react-native-modal-datetime-picker";
+import { ArrowRight } from "lucide-react-native";
+
+export default function EventCreation() {
+  const form = useForm({
+    defaultValues: {
+      title: "",
+      address: "",
+      description: "",
+      eventDate: new Date(),
+    },
+  });
+
+  const createEvent = useEventCreation();
+  const { data: user, isLoading } = useUser();
+
+  const [isDatePickerVisible, setDatePickerVisibility] = useState(false); //for date picker
+  const [isTimePickerVisible, setTimePickerVisibility] = useState(false); //for time picker
+
+  const onSubmit = form.handleSubmit(async ({ eventDate, ...data }) => {
+    if (!user) {
+      alert("You must be logged in to create an event.");
+      return;
+    }
+
+    try {
+      await createEvent.mutateAsync({ eventDate: eventDate.toUTCString(), ...data, organizer: user.id });
+      router.push("/yourEvents");
+    } catch (error) {
+      alert("You are not allowed to perform this action.");
+    }
+  });
+
+  if (isLoading) {
+    return <SizableText>Loading...</SizableText>;
+  }
+
+  // controller for the date
+  const showDatePicker = () => {
+    setDatePickerVisibility(true);
+  };
+
+  const hideDatePicker = () => {
+    setDatePickerVisibility(false);
+  };
+
+  // controller for the time
+  const showTimePicker = () => {
+    setTimePickerVisibility(true);
+  };
+
+  const hideTimePicker = () => {
+    setTimePickerVisibility(false);
+  };
```

Comment:
Personally, I do prefer just doing these inline, but I am also fine with this. Just you know, preferences 😅